src/test/ui/lint/lint-unsafe-code.stderr

```diff
@@ -1,89 +1,168 @@
-error: declaration of an `unsafe` function
-  --> $DIR/lint-unsafe-code.rs:23:1
+error: declaration of a `no_mangle` function
+  --> $DIR/lint-unsafe-code.rs:31:17
    |
-LL | unsafe fn baz() {}
-   | ^^^^^^^^^^^^^^^^^^
+LL | #[no_mangle] fn foo() {}
```

Comment:
Could the lint span point to the attribute itself, rather than the identifier of the item being defined?